cri/v1alpha1/cri_utils.go

```diff
@@ -940,3 +941,158 @@ func blkrdev(device string) (uint64, error) {
+	var memorySwappiness *runtime.Int64Value
+	if apitypesResources.MemorySwappiness != nil {
+		memorySwappiness = &runtime.Int64Value{Value: *apitypesResources.MemorySwappiness}
+	}
+
+	return &runtime.LinuxContainerResources{
+		CpuPeriod:             apitypesResources.CPUPeriod,
+		CpuQuota:              apitypesResources.CPUQuota,
+		CpuShares:             apitypesResources.CPUShares,
+		MemoryLimitInBytes:    apitypesResources.Memory,
+		CpusetCpus:            apitypesResources.CpusetCpus,
+		CpusetMems:            apitypesResources.CpusetMems,
+		BlkioWeight:           uint32(apitypesResources.BlkioWeight),
+		BlkioWeightDevice:     parseWeightDeviceFromPouch(apitypesResources.BlkioWeightDevice),
+		BlkioDeviceReadBps:    parseThrottleDeviceFromPouch(apitypesResources.BlkioDeviceReadBps),
+		BlkioDeviceWriteBps:   parseThrottleDeviceFromPouch(apitypesResources.BlkioDeviceWriteBps),
+		BlkioDeviceRead_IOps:  parseThrottleDeviceFromPouch(apitypesResources.BlkioDeviceReadIOps),
+		BlkioDeviceWrite_IOps: parseThrottleDeviceFromPouch(apitypesResources.BlkioDeviceWriteIOps),
+		KernelMemory:          apitypesResources.KernelMemory,
+		MemoryReservation:     apitypesResources.MemoryReservation,
+		MemorySwappiness:      memorySwappiness,
+		Ulimits:               parseUlimitFromPouch(apitypesResources.Ulimits),
+		DiskQuota:             diskQuota,
+	}
+}
+
+// parseWeightDeviceFromAPI parse WeightDevice from runtime.WeightDevice to apitypes.WeightDevice
+func parseWeightDeviceFromAPI(runtimeWeightDevices []*runtime.WeightDevice) (weightDevices []*apitypes.WeightDevice) {
+	if runtimeWeightDevices == nil {
+		return nil
+	}
+	for _, v := range runtimeWeightDevices {
+		weightDevices = append(weightDevices, &apitypes.WeightDevice{
+			Path:   v.GetPath(),
+			Weight: uint16(v.GetWeight()),
+		})
+	}
+	return weightDevices
+}
+
+// parseWeightDeviceFromPouch parse WeightDevice from apitypes.WeightDevice to runtime.WeightDevice
+func parseWeightDeviceFromPouch(apitypesWeightDevices []*apitypes.WeightDevice) (weightDevices []*runtime.WeightDevice) {
+	if apitypesWeightDevices == nil {
+		return nil
+	}
+	for _, v := range apitypesWeightDevices {
+		weightDevices = append(weightDevices, &runtime.WeightDevice{
+			Path:   v.Path,
+			Weight: uint32(v.Weight),
+		})
+	}
+	return weightDevices
+}
+
+// parseThrottleDeviceFromAPI parse ThrottleDevice from runtime.WeightDevice to apitypes.WeightDevice
+func parseThrottleDeviceFromAPI(runtimeThrottleDevices []*runtime.ThrottleDevice) (throttleDevices []*apitypes.ThrottleDevice) {
+	if runtimeThrottleDevices == nil {
+		return nil
+	}
+	for _, v := range runtimeThrottleDevices {
+		throttleDevices = append(throttleDevices, &apitypes.ThrottleDevice{
+			Path: v.GetPath(),
+			Rate: v.GetRate(),
+		})
+	}
+	return throttleDevices
+}
+
+// parseThrottleDeviceFromPouch parse ThrottleDevice from apitypes.WeightDevice to runtime.WeightDevice
```

Comment:
The `WeightDevice` in comment should be `ThrottleDevice`?

Reply:
I'm sorry for my carelessness. it has been corrected.